### PR TITLE
chore!: audit corrections

### PR DESCRIPTION
**Pull request by brianp**
Description
---
- Updates one integer casting to be safer
- Changes a function privacy as per the docs
- Removes outdated comments
- Makes domain hashes consistent in the repo

Motivation and Context
---
Minor changes for the audit, mostly in the documentation. 

How Has This Been Tested?
---
CI

Breaking Changes
---

- [ ] None
- [ ] Requires data directory on base node to be deleted
- [ ] Requires hard fork
- [x] Other - Please specify

Message hash changes will make messages between older nodes incompatible with messages sent by newer nodes.

### CODE REVIEW

File: base_layer/core/src/transactions/key_manager/inner.rs [hunk 1]
```diff
@@ -84,7 +84,10 @@ use crate::{
     },
 };
 
-hash_domain!(KeyManagerHashingDomain, "base_layer.core.key_manager");
+hash_domain!(
+    KeyManagerHashingDomain,
+    "com.tari.base_layer.core.transactions.key_manager"
+);
```
Comment (AaronFeickert) [inline base_layer/core/src/transactions/key_manager/inner.rs:90]: I believe this introduces a breaking change, which should be noted in the PR title and description. But good catch.

File: base_layer/core/src/transactions/transaction_components/side_chain/mod.rs [hunk 2]
```diff
@@ -36,15 +36,15 @@ pub use validator_node_signature::{ValidatorNodeHashDomain, ValidatorNodeSignatu
 
 hash_domain!(
     ContractAcceptanceHashDomain,
-    "com.tari.tari-project.base_layer.core.transactions.side_chain.contract_acceptance_challenge",
+    "com.tari.base_layer.core.transactions.side_chain.contract_acceptance_challenge",
```
Comment (AaronFeickert) [inline base_layer/core/src/transactions/transaction_components/side_chain/mod.rs:39]: Also a breaking change.

File: base_layer/core/src/transactions/transaction_components/side_chain/mod.rs [hunk 3]
```diff
@@ -36,15 +36,15 @@ pub use validator_node_signature::{ValidatorNodeHashDomain, ValidatorNodeSignatu
...
     1
 );
 
 pub type ContractAcceptanceHasherBlake256 = DomainSeparatedHasher<Blake256, ContractAcceptanceHashDomain>;
 
 hash_domain!(
     SignerSignatureHashDomain,
-    "com.tari.tari-project.base_layer.core.transactions.side_chain.signer_signature",
+    "com.tari.base_layer.core.transactions.side_chain.signer_signature",
```
Comment (AaronFeickert) [inline base_layer/core/src/transactions/transaction_components/side_chain/mod.rs:47]: Also also a breaking change.

File: base_layer/core/src/transactions/transaction_components/side_chain/validator_node_signature.rs [hunk 4]
```diff
@@ -27,7 +27,11 @@ use tari_common_types::types::{FixedHash, PrivateKey, PublicKey, Signature};
 use tari_crypto::{hash::blake2::Blake256, hash_domain, hashing::DomainSeparatedHasher, keys::PublicKey as PublicKeyT};
 use tari_utilities::ByteArray;
 
-hash_domain!(ValidatorNodeHashDomain, "com.tari.dan_layer.validator_node", 0);
+hash_domain!(
+    ValidatorNodeHashDomain,
+    "com.tari.base_layer.core.transactions.side_chain.validator_node",
+    0
+);
```
Comment (AaronFeickert) [inline base_layer/core/src/transactions/transaction_components/side_chain/validator_node_signature.rs:34]: Also also also a breaking change.

File: base_layer/core/src/transactions/transaction_protocol/mod.rs [hunk 5]
```diff
@@ -188,7 +188,7 @@ pub struct RecoveryData {
 // hash domain
 hash_domain!(
     CalculateTxIdTransactionProtocolHashDomain,
-    "com.tari.tari-project.base_layer.core.transactions.transaction_protocol.calculate_tx_id",
+    "com.tari.base_layer.core.transactions.transaction_protocol.calculate_tx_id",
```
Comment (AaronFeickert) [inline base_layer/core/src/transactions/transaction_protocol/mod.rs:191]: See above.

**Review by AaronFeickert — APPROVED** [review]
Looks good except for updating the PR metadata to indicate breaking changes introduced by the hash domain updates.

**Comment by AaronFeickert** [comment]
~~Looks like CI is a bit borked at the moment, but it should be asserted that no unit tests fail due to the hash domain changes, in case of out-of-date test vectors.~~

Looks like CI is all good!

File: base_layer/core/src/transactions/key_manager/inner.rs (hunk 1 above)
Comment (brianp) [reply base_layer/core/src/transactions/key_manager/inner.rs:90]: Totally correct. I had thought about it when making the initial change but forgot while pushing up the PR. I figure now might as well be the right time considering the change is entirely about consistency, breaking things now is better than later.

Comment (AaronFeickert) [reply base_layer/core/src/transactions/key_manager/inner.rs:90]: For sure; consistent naming reduces the chance of unintentional domain collisions.

**Review by SWvheerden — APPROVED** [review]
Ack

**Pull request merged**